### PR TITLE
Eliminate async

### PR DESCRIPTION
**Pull request by Danp2**

### CODE REVIEW

File: library.cpp [hunk 1]
```diff
@@ -140,6 +140,67 @@ namespace /*anon*/ {
...
+            }));
+
+            // Perform the websocket handshake
+
+            // Host HTTP header includes the port. See https://tools.ietf.org/html/rfc7230#section-5.4
+            ws_.handshake(utf8_encode(host_) + ":" + utf8_encode(port_), utf8_encode(path_)); 
+
+            // Send the message
+            VERBOSE(L"Issue async_read in on_handshake");
+            ws_.async_read(buffer_, beast::bind_front_handler(&Session::on_read, shared_from_this()));
```
Comment (sehe) [inline library.cpp:199]: Mixing async and sync is not a feature in the Beast library. Basically, their internal operations have state that assume it's either sync or async.

**Comment by sehe** [comment]
I have committed my suggested interface changes here: https://github.com/Danp2/WebSocketAsio/pull/11

File: library.cpp (hunk 1 above)
Comment (Danp2) [reply library.cpp:199]: Ok. I was thinking that the above was needed to kick off the initial read. 

> I have committed my suggested interface changes here: #11

I was actually starting to work on this myself, so your timing is perfect. I'll take a look at this in a bit.

Comment (sehe) [reply library.cpp:199]: > Ok. I was thinking that the above was needed to kick off the initial read.

Oh it was. However, you cannot mix synchronous websockets methods with that

Comment (sehe) [reply library.cpp:199]: > Ok. I was thinking that the above was needed to kick off the initial read.

Oh it was. However, you cannot mix synchronous websockets methods with that

**Pull request closed without merging**